src/main/java/use_case/preferences/PreferenceInputData.java

```diff
@@ -1,36 +1,81 @@
 package use_case.preferences;
 
 import java.util.Date;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 /**
  * The Input Data for the Signup Use Case.
  */
 public class PreferenceInputData {
-
+    private final String fullname;
     private final String username;
+    private final String password;
+    private final String image;
+    private final String repeatPassword;
+    private final String location;
+    private final String gender;
+    private final Date dateOfBirth;
+    private final List<String> preferredGender;
+    private final HashMap<String, Integer> preferredAge;
     private final List<String> tags;
     private final String bio;
     private final Map<String, Boolean> preferences;
-    private final List<String> preferredGender;
-    private final Map<String, Integer> preferredAge;
 
-    public PreferenceInputData(String username, List<String> tags, String bio,  Map<String, Boolean> preferences,
-                               List<String> preferredGender, Map<String, Integer> preferredAge) {
+    public PreferenceInputData(String fullname, String username, String password, String image, String repeatPassword,
+                               String location, String gender, Date dateOfBirth, List<String> preferredGender,
```

Comment:
Correctness: A good review for us that the order of instance variables in a class does NOT matter when creating a constructor. Only the ORDER of parameters in the constructor matter.